src/Migration/Run/RunService.php

```diff
@@ -134,19 +136,42 @@ public function getRunStatus(Context $context): MigrationState
     }
 
     /**
-     * @param array<int, string>|null $credentialFields
+     * @param array<string, mixed>|null $credentialFields
      */
     public function updateConnectionCredentials(Context $context, string $connectionUuid, ?array $credentialFields): void
     {
         if ($this->isMigrationRunning($context)) {
             throw MigrationException::migrationIsAlreadyRunning();
         }
 
-        $context->scope(MigrationContext::SOURCE_CONTEXT, function (Context $context) use ($connectionUuid, $credentialFields): void {
+        /** @var SwagMigrationConnectionEntity $connection */
+        $connection = $this->connectionRepo->search(new Criteria([$connectionUuid]), $context)->first();
```

Comment:
```suggestion
        $connection = $this->connectionRepo->search(new Criteria([$connectionUuid]), $context)->getEntities()->first();
```
could be that you can remove the `@var` annotation with this